docs/setup_vscode_wsl.md

```diff
@@ -0,0 +1,575 @@
+
+```console
+$ code --version
+1.74.1
+1ad8d514439d5077d2b0b7ee64d2ce82a9308e5a
+x64
+```
+
+### Extensions
+Make sure VS Code is installed correctly by checking the version.  You need version 1.52.1 or higher.
+```console
+$ code --version
+1.52.1
+```
+
+Install the Microsoft [C/C++ extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) and the [WSL extension](https://code.visualstudio.com/docs/remote/wsl).  See the [C/C++ extension alternatives](#cc-extension-alternatives) section for details about why we recommend these extensions.
+```console
+$ code --install-extension ms-vscode.cpptools
+$ code --install-extension ms-vscode-remote.remote-wsl
+```
+
+Restart VS Code.
+
+Verify that the extension is installed.  It's OK if you have other extensions installed.
+```console
+$ code --list-extensions
+ms-vscode.cpptools
+ms-vscode-remote.remote-wsl
+```
+
+## Create a project
+To create a VS Code project, create a folder (directory).  There are many ways to create folders: Finder AKA File Explorer, VS Code interface, VS Code integrated terminal, and the system terminal.  We'll use the system terminal and call our example project `p1-stats`.
+
+Open the Terminal ([Ubuntu Bash Shell](cli.html#open-terminal-windows)).
+
+Navigate to your home directory, create a new directory, then move into the new directory. Your folder location might be different.  Here's some help with [`cd`](cli.html#cd), the [tilde `~`](cli.html#home-directory-), and [`mkdir`](cli.html#mkdir).
+
+```console
+$ mkdir ~/eecs280
+$ cd ~/eecs280
+$ mkdir p1-stats
+$ cd p1-stats
+```
+
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Avoid paths that contain spaces.  Spaces causes problems with some command line tools.
+
+| Bad Example     | Good Example   |
+|-----------------|----------------|
+| `EECS 280/` | `eecs280/` |
+| `Project 1 Stats/` | `p1-stats/` |
+
+</div>
+  
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Linux (Ubuntu) has a separate home directory.  Storing code in your Windows home directory can cause slowdowns.
+
+```console
+$ pwd
+/home/awdeorio ...         # Good, Linux home
+/c/mnt/Users/awdeorio ...  # Bad, Windows home
+```
+
+Here's how to [access your Linux files from Windows](setup_wsl.html#accessing-linux-files-from-windows).
+
+</div>
+
+Start VS Code and open your project folder by selecting `File` > `Open Folder...` > navigate to the `p1-stats` folder.
+
+<div class="primer-spec-callout info" markdown="1">
+**Pro-tip:** Here's a quick way to open VS Code to a specific project folder from the command line.  First make sure you're in the directory that contains your source code.
+```console
+$ ls
+main.cpp ...
+$ code .
+```
+</div>
+
+<img src="images/vscode020.png" width="768px" />
+
+### Add new files
+Open your project folder by selecting `File` > `Open Folder...` > navigate to the `p1-stats` folder.
```

Comment:
I think this "open your project folder" part can be removed since it was just covered immediately above. (If it is kept, it needs to be fixed in the same way as the above to use "open folder in WSL" instead.)

Reply:
The reason this is here is for students visiting the section via a fragment link from a project spec, e.g.,  https://eecs280staff.github.io/tutorials/setup_vscode.html#add-new-files

Reply:
Uh oh.  I just realized that this PR will break all the fragment links into the VSCode tutorial!

Reply:
Update: I create an issue in each project repo to address this problem.

Reply:
I don't have a strong preference, but it will need to be adjusted to the "open folder in WSL" instructions as with the previous instance.

Reply:
I'll make a separate issue for this fix so that we can get the PR through.